sys/arm64/include/intr.h

```diff
@@ -27,6 +27,7 @@
 #ifndef _MACHINE_INTR_H_
 #define	_MACHINE_INTR_H_
 
+#ifndef LOCORE
 #ifdef FDT
 #include <dev/ofw/openfirm.h>
 #endif
```

Comment:
```suggestion
#endif

typedef enum {
	INTR_ROOT_IRQ =	_INTR_ROOT_IRQ,
	INTR_ROOT_FIQ =	_INTR_ROOT_FIQ,
	INTR_ROOT_END
} intr_root_t;
#define INTR_ROOT_COUNT	INTR_ROOT_END
```
This allows the C constructs to handle this as an enumerated value (which it amounts to), while allowing the assembly language portion to still have its values.

Reply:
I considered it but I'm not a fan of this approach. The problem is that you're setting the enum values and also depending on `INTR_ROOT_END` having the correct value simply because it's at the end. For more complicated cases this seems error-prone.

Reply:
Hrmm, seems I've discovered unexpected behavior in how GitHub handles comments...

As such replacing the mention.  This is no more error-prone than demanding list of functions end with `{0, 0}` or `DEVMETHOD_END`.  Yes, it is possible to goof this, but it is quite easy to get it right.  This isn't severe enough to worry.  Take a look at my proposed examples.